### DISCUSSION TITLE
`@responsive` not working when used inside an imported css file

### DISCUSSION THREAD

**Discussion by Sparragus** [tailwindlabs/tailwindcss#4528] [Help] [answered]
Hi everyone. And to the authors and contributors of TailwindCSS, thank you! <3

I'm having issues using `@responsive` inside an imported css.

Here's a NextJS project reproducing the issue: https://codesandbox.io/s/tailwind-responsive-issue-2squ8?file=/styles/buttons.css

In summary, I have two files: `index.css` and `buttons.css`.

```css
/* index.css */
@tailwind base;
@tailwind components;
@tailwind utilities;

@import "./buttons.css";
```

```css
/* buttons.css */
.btn {
  @apply px-2.5 py-2.5 rounded inline-flex items-center text-sm font-normal whitespace-nowrap outline-none cursor-pointer;
}

.btn-black {
  @apply text-white bg-black;
}

@responsive {
  .btn-red {
    @apply text-white bg-red-500;
  }
}
```

After importing `index.css` inside `_app.js`, I use the classes like so:

```js
// index.js
import * as React from "react";

export default function Buttons() {
  return (
    <div className="p-8">
      <h1 className="mb-12 text-4xl">Buttons</h1>

      <h2 className="mb-8 text-2xl">Default button</h2>

      <div className="flex space-x-8">
        <button className="btn btn-black">Black button</button>

        <button className="btn btn-red">Red button</button>
      </div>
    </div>
  );
}
```

The black button does render as expected, but the red button doesn't.

![image](https://user-images.githubusercontent.com/80982/120348213-98ec5080-c2c2-11eb-8fa5-8ab4ce7193ca.png)

Furthermore, if I wrap the css code inside `buttons.css` with `@layer components { ... }`, none of the button styles will work.

Any ideas of what I'm missing or may be doing wrong? 

Much appreciated. :)

**Comment by Sparragus** [comment, marked as answer]
I just solved my own question....two minutes later. 🙃 

If you run into this issue, you may need to use the [`postcss-import`](https://github.com/postcss/postcss-import) plugin.

To fix:

1. Install [`postcss-import`](https://github.com/postcss/postcss-import#installation)
2. Add the plugin inside your `postcss.config.js`. My config looks like:
```js
module.exports = {
  plugins: {
    "postcss-import": {},
    tailwindcss: {},
    autoprefixer: {},
  },
};
```
3. Replace the `@tailwind` directive with `@import`.
```css
@import "tailwindcss/base";
@import "tailwindcss/components";
@import "tailwindcss/utilities";
@import "./buttons.css";
```
4. Restart your NextJS server.
5. It should be working.

The solution is also described here: https://tailwindcss.com/docs/using-with-preprocessors#build-time-imports

**Reply by simonswiss** [reply]
That's the way! ☝️
